service/discovery/azure/storage.go

```diff
@@ -523,3 +378,35 @@ func accountName(id string) string {
 	splitName := strings.Split(id, "/")
 	return splitName[8]
 }
+
+// initializeBlobContainerClient creates the client if not already exists
+func (d *azureStorageDiscovery) initializeBlobContainerClient() (err error) {
```

Comment:
I think we just used `initXXX` somewhere else

Reply:
You mean the naming or do you wanted to use the other generic function? 

Is it possible to change these methods to generic methods?

Reply:
I was referring to the naming. Would be interesting if these could be generic functions.